shell/components/nav/TopLevelMenu.vue

```diff
@@ -29,6 +33,17 @@ export default {
     const { displayVersion, fullVersion } = getVersionInfo(this.$store);
     const hasProvCluster = this.$store.getters[`management/schemaFor`](CAPI.RANCHER_CLUSTER);
 
+    const canPagination = this.$store.getters[`management/paginationEnabled`]({
+      id:      MANAGEMENT.CLUSTER,
+      context: 'side-bar',
+    }) && this.$store.getters[`management/paginationEnabled`]({
+      id:      CAPI.RANCHER_CLUSTER,
+      context: 'side-bar',
+    });
+    const helper = canPagination ? new TopLevelMenuHelperPagination({ $store: this.$store }) : new TopLevelMenuHelperLegacy({ $store: this.$store });
```

Comment:
Not for this PR but I have a feeling we're going to want to make these helpers more generic and perhaps add the updating/debouncing to the thelpers

Reply:
That's my plan for the PaginationWrappers (very open to renaming, maybe PaginationHandler?). Those should be able to receive some pagination settings, return the page and also notify when that page has changed. It's a new util though so not wired in everywhere